akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala

```diff
@@ -228,6 +229,56 @@ class ConnectionPoolSpec extends AkkaSpec("""
       acceptIncomingConnection()
       val (Success(_), 42) = responseOut.expectNext()
     }
+
+    "never close hot connections when minConnections key is given and >0 (minConnections = 1)" in new TestSetup() {
+      val close: HttpHeader = Connection("close")
+
+      // for lower bound of one connection
+      val minOneConnection = 1
+      val (requestInOne, requestOutOne, responseOutSubOne, hcpMinOneConnection) =
+        cachedHostConnectionPool[Int](idleTimeout = 100.millis, minConnections = minOneConnection)
+      val gatewayOneConnection = hcpMinOneConnection.gateway
+
+      acceptIncomingConnection()
+      requestInOne.sendNext(HttpRequest(uri = "/minimumslots/1", headers = immutable.Seq(close)) → 42)
+      responseOutSubOne.request(1)
+      requestOutOne.expectNextN(1)
+
+      condHolds(500.millis) { () ⇒
+        Await.result(gatewayOneConnection.poolStatus(), 100.millis).get shouldBe a[PoolInterfaceRunning]
+      }
+    }
+
+    "never close hot connections when minConnections key is given and >0 (minConnections = 5)" in new TestSetup() {
+      val close: HttpHeader = Connection("close")
+
+      // for lower bound of five connections
+      val minFiveConnections = 5
+      val (requestInFive, requestOutFive, responseOutSubFive, hcpMinFiveConnection) = cachedHostConnectionPool[Int](
+        idleTimeout = 100.millis,
+        minConnections = minFiveConnections,
+        maxConnections = minFiveConnections + 10)
```

Comment:
It could be any number >0. The default for tests is 3, which would cause an error (min > max)